File: LeetcodeQuestions/non_decreasing_arr-665.rb
# @param {Integer[]} nums
# @return {Boolean}
def check_possibility(nums)
    pos  = -1
    (0..nums.length - 2).each do |i|
        if nums[i] > nums[i + 1]
            return false if pos != -1
            pos = i
        end
    end
    return pos == -1 || pos == 0 || pos == nums.length - 2 || nums[pos - 1] <= nums[pos + 1] || nums[pos] <= nums[pos + 2]
end

File: LeetcodeQuestions/non_decreasing_arr-665.py
class Solution(object):
    def checkPossibility(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        pos = -1
        for i in range(len(nums) - 1):
            if nums[i] > nums[i + 1]:
                if pos != -1:
                    return False
                pos = i
        return pos == -1 or pos == 0 or pos == len(nums) - 2 or nums[pos - 1] <= nums[pos + 1] or nums[pos] <= nums[pos + 2]
